File: PSET1/mario.c
//this program asks for the height of a Nintendo Mario pyramid from the user and then creates it using hashes.
#include <cs50.h>
#include <stdio.h>

int get_positive_int(string prompt);

int main(void)
{
    //asking user for the height
    int height = get_positive_int("Height: ");
    //This is the main loop loop that handle the height of each pyramid
    for (int i = 0; i < height; i++)
    {
        //this loop uses the given height minus one to create the spaces needed to justify/align the hashes to the right
        for (int k = 1; k < height - i; k++)
        {
            printf(" ");
        }
        //this loop prints the amount of hashes for each row based on each iteration for the given height
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }
        //this creates a new line after each iteration of the outer for loop
        printf("\n");
       
    }
    
}

// Prompt user for positive integer
int get_positive_int(string prompt)
{
    int n;
    do
    {
        n = get_int("%s", prompt);
    }     
    while (n > 8 || n < 1);
    return n;
}

File: PSET6/mario.py
# this program creates a nintendo mario like pyramid by first asking the user for the height. And the pyramid blocks are represented by hashes
import cs50

# if the input is not a number greater than zero or less than 8 inclusive, the user will be prompted until a number from 1-8 is entered
while True:
    height = cs50.get_int("height: ")
    if height > 0 and height < 9:
        break
# this for loop will start at zero up to the value of height +1 to get desired amount of blocks
for x in range(0, height+1, 1):
    print(x * "#")
    print(((height - x)-1) * " ", end="")
